### DISCUSSION TITLE
Multiple [Feature Requests]

### DISCUSSION THREAD

**Discussion by Terkyz** [padloc/padloc#556] [Ideas]
These are some functions that, in my opinion, can improve the experience and usability of Padloc, the order in which they are written is not an order of importance, and maybe they are already under development, personally I quickly checked on Github and I didn't find anything.

1. Pin: Ability to Pin a vault, with "Pin" I mean that the Vault that I choose is placed first in my general list of Vaults, without respecting the order in which they are placed, why? I personally use some services that require frequent use of the TOTP, and it is a bit boring to search between my favorites list or my Vault, I think that this process could be speeded up if the Pin option existed, since mainly I only need to access that specific TOTP

2. Move fields from a vault using a Drag and drop capability: As the name suggests, I think editing a Vault can also be speeded up, and this can be done by adding the ability to move fields from a vault not just using the arrow keys. up or down next to my field, but grab the entire field and move it to the desired position, and in fact trying this I realized that you can grab a field outside the edit option, but it does nothing

3. Log-in alternatives: On mobile devices, it is relatively simple to access without having to enter my password every time (using the biometric function), but on PC it is not like that, and it would be great if there were alternative login options, such as attaching a file to work as an "access key", or enter a pin.

4. Custom Homepage: Ability to define the home screen when starting the program, for example, that when starting Padloc, by default I could choose to open it in my Vault with the "TOTP" tag and not in "My Vault".

Thank you! I can open an Issue for each of these functions but I prefer to keep it simple

**Comment by MaKleSoft** [comment]
Thanks for you feedback, @Terkyz!

> Pin: Ability to Pin a vault, with "Pin" I mean that the Vault that I choose is placed first in my general list of Vaults, without respecting the order in which they are placed, why? I personally use some services that require frequent use of the TOTP, and it is a bit boring to search between my favorites list or my Vault, I think that this process could be speeded up if the Pin option existed, since mainly I only need to access that specific TOTP

Just to make sure we are talking about the same thing - are you perhaps talking about vault **items**? If I understand you correctly, you want to make it so that certain items always appear at the top of the list, correct?
Have you checked out the [recently used](https://docs.padloc.app/manual/search/#recently-used-items) feature? Sounds like it might help you with your use case? 

> Move fields from a vault using a Drag and drop capability: As the name suggests, I think editing a Vault can also be speeded up, and this can be done by adding the ability to move fields from a vault not just using the arrow keys. up or down next to my field, but grab the entire field and move it to the desired position, and in fact trying this I realized that you can grab a field outside the edit option, but it does nothing

The problem with using drag & drop for reordering is that while it works great for pointer-based inputs (mouse or trackpad) it breaks down completely on touch devices.  Another issue is general discoverability and, more importantly, **accessibility**: Up- and down arrows, while slightly less convenient to use than drag & drop, work reliably for all input types, don't require an explanation, and are easier to make accessible to things like screen readers. Overall, since it's a feature that (I think) is used fairly rarely we opted the more "robust" implementation over the more convenient one.

> Log-in alternatives: On mobile devices, it is relatively simple to access without having to enter my password every time (using the biometric function), but on PC it is not like that, and it would be great if there were alternative login options, such as attaching a file to work as an "access key", or enter a pin.

This is a thing that comes up quite often. We do want to strike a good balance between convenience and security, but any kind of "quick unlock" feature comes with a ton of security-related challenges that are anything but trivial to solve. Having said that, we do plan to bring biometric unlock to as many platforms as possible, and find other forms of "quick unlock" where biometric sensors are not available. Stay tuned 😉

> Custom Homepage: Ability to define the home screen when starting the program, for example, that when starting Padloc, by default I could choose to open it in my Vault with the "TOTP" tag and not in "My Vault".

This is already tracked here: https://github.com/padloc/padloc/issues/518

> Thank you! I can open an Issue for each of these functions but I prefer to keep it simple

In the future, please do create separate tickets for each distinct feature request. This make it much easier for us to triage and discuss issues. If you have some vague ideas for features or want to discuss the merit of an idea before creating a specific feature request, please post your thoughts under [discussions](https://github.com/padloc/padloc/discussions).

**Reply by Terkyz** [reply]
> Thanks for you feedback, @Terkyz!
> 
> > Pin: Ability to Pin a vault, with "Pin" I mean that the Vault that I choose is placed first in my general list of Vaults, without respecting the order in which they are placed, why? I personally use some services that require frequent use of the TOTP, and it is a bit boring to search between my favorites list or my Vault, I think that this process could be speeded up if the Pin option existed, since mainly I only need to access that specific TOTP
> 
> Just to make sure we are talking about the same thing - are you perhaps talking about vault **items**? If I understand you correctly, you want to make it so that certain items always appear at the top of the list, correct? Have you checked out the [recently used](https://docs.padloc.app/manual/search/#recently-used-items) feature? Sounds like it might help you with your use case?
> 
I did not know that, next time I will read the documentation, also as I migrated from Keepass to Padloc I had to manually edit all the entries and they all appear in "recently used", but now understood how it works!
> > Move fields from a vault using a Drag and drop capability: As the name suggests, I think editing a Vault can also be speeded up, and this can be done by adding the ability to move fields from a vault not just using the arrow keys. up or down next to my field, but grab the entire field and move it to the desired position, and in fact trying this I realized that you can grab a field outside the edit option, but it does nothing
> 
> The problem with using drag & drop for reordering is that while it works great for pointer-based inputs (mouse or trackpad) it breaks down completely on touch devices. Another issue is general discoverability and, more importantly, **accessibility**: Up- and down arrows, while slightly less convenient to use than drag & drop, work reliably for all input types, don't require an explanation, and are easier to make accessible to things like screen readers. Overall, since it's a feature that (I think) is used fairly rarely we opted the more "robust" implementation over the more convenient one.
> 
> > Log-in alternatives: On mobile devices, it is relatively simple to access without having to enter my password every time (using the biometric function), but on PC it is not like that, and it would be great if there were alternative login options, such as attaching a file to work as an "access key", or enter a pin.
> 
> This is a thing that comes up quite often. We do want to strike a good balance between convenience and security, but any kind of "quick unlock" feature comes with a ton of security-related challenges that are anything but trivial to solve. Having said that, we do plan to bring biometric unlock to as many platforms as possible, and find other forms of "quick unlock" where biometric sensors are not available. Stay tuned wink
> 
This makes me think of something a bit off topic, when I exported my Keepass files from .keepass to .csv Keepass warned me that "it is a less secure format", how true is that? I guess Padloc will use its own encrypted format once I import it
> > Custom Homepage: Ability to define the home screen when starting the program, for example, that when starting Padloc, by default I could choose to open it in my Vault with the "TOTP" tag and not in "My Vault".
> 
> This is already tracked here: #518
> 
> > Thank you! I can open an Issue for each of these functions but I prefer to keep it simple
> 
> In the future, please do create separate tickets for each distinct feature request. This make it much easier for us to triage and discuss issues. If you have some vague ideas for features or want to discuss the merit of an idea before creating a specific feature request, please post your thoughts under [discussions](https://github.com/padloc/padloc/discussions).

I didn't know that, I just started using Github and honestly I think this "Discussions" are very useful, I'll keep using it!

**Comment by Terkyz** [comment]
I was thinking about another function: 

automatic local backup
instead of having to manually save my database, padloc allows me to define a path in my system where it perform periodic database backups to prevent data loss, extremely useful to combine it with a cloud storage.

and function two, although really this is something more subjective and not necessary: 

​​automatic saving just after making any change in a vault
editing multiple vaults at the same time has led me several times to have synchronization problems between devices, since the vault is not saved directly at once but waits for a certain time, it would be great having the to perform a sync save just before editing a vault or some value.

**Reply by MaKleSoft** [reply]
> ​​automatic saving just after making any change in a vault
editing multiple vaults at the same time has led me several times to have synchronization problems between devices, since the vault is not saved directly at once but waits for a certain time, it would be great having the to perform a sync save just before editing a vault or some value.

Can you elaborate on this? How are you "editing multiple vaults at the same time"? Are you using multiple devices simultaneously? Also, I assume you are talking about "Vault Items", or entries within a Vault, not the Vaults themselves, correct? What are the "synchronization problems" you are referring to? Is data being overwritten, deleted, not being saved? The more details you can give us the better :)

**Reply by Terkyz** [reply]
> > ​​automatic saving just after making any change in a vault
> > editing multiple vaults at the same time has led me several times to have synchronization problems between devices, since the vault is not saved directly at once but waits for a certain time, it would be great having the to perform a sync save just before editing a vault or some value.
> 
> Can you elaborate on this? How are you "editing multiple vaults at the same time"? Are you using multiple devices simultaneously? Also, I assume you are talking about "Vault Items", or entries within a Vault, not the Vaults themselves, correct? What are the "synchronization problems" you are referring to? Is data being overwritten, deleted, not being saved? The more details you can give us the better :)

With "editing multiple Vault [Items] at the same time" I meant: When I edit a Vault Item and immediately edit another, without giving Padloc enough time to perform its automatic save!

When I do that, Padloc doesn't save the changes from the first edit, only from the second edit, the data from the first edit just doesn't change!

I was trying to replicate this behavior in the desktop version of Padloc and it doesn't happen, as the save happens immediately! So it's an Android problem, but I would accuse the performance of my phone in Padloc, since it lags when I'm inside Padloc

**Reply by MaKleSoft** [reply]
> When I do that, Padloc doesn't save the changes from the first edit, only from the second edit, the data from the first edit just doesn't change!

That definitely sounds like a bug. Would you mind creating an issue for that?

**Reply by Terkyz** [reply]
Is it possible to implement a button to "ignore" Securities Reports of specific Vaults Item?

**Reply by MaKleSoft** [reply]
@Terkyz That's a great idea! I created a ticket for it: https://github.com/padloc/padloc/issues/566

**Comment by Terkyz** [comment]
I think about another feature that may be useful for PC users (if this does not affect the accessibility of the program) and that is to scroll horizontally between the details of a vault item using the mouse scroll!

**Reply by BrunoBernardino** [reply]
Could you elaborate @Terkyz ? This should already be possible in the item list view. Where are you expecting it to work that it doesn't work?

**Reply by Terkyz** [reply]
When I try to scroll horizontally in the details of a Vault Item, nothing happens.

When I try to do [this](https://imgur.com/a/r7i7PdA), [here](https://imgur.com/a/YKPlTGA)

**Reply by BrunoBernardino** [reply]
Thanks for the gifs @Terkyz ! Are you using a mouse/scrollwheel? I have no trouble with a trackpad, scrolling horizontally, but haven't tested with a mouse yet.

Also, if you click on the right-way-pointing-arrow that shows up when you hover the item, does that scroll?

**Reply by Terkyz** [reply]
> Thanks for the gifs @Terkyz ! Are you using a mouse/scrollwheel? I have no trouble with a trackpad, scrolling horizontally, but haven't tested with a mouse yet.
> 
> Also, if you click on the right-way-pointing-arrow that shows up when you hover the item, does that scroll?

Yep, I'm using my mouse's scrollwhell.

Press the arrow if you scroll.

Here another [scroll test video](https://imgur.com/2l9sU7Z)

**Reply by BrunoBernardino** [reply]
Ok, thanks for showing that off! That looks like it's working as expected, then, because if scrolling vertically there scrolled that part horizontally, once you had more items, it would be impossible to scroll through them vertically.

To scroll horizontally there, you need to be intentionally scrolling horizontally (with a trackpad, for example, or perhaps there's a shift or ctrl + mousewheel combo that also works, depending on your setup/shortcuts).

I hope that makes sense!

**Comment by Terkyz** [comment]
After the update that implemented the "Expirations", it is possible to add an ability to update a password automatically once that expiration date is met, and send a reminder email to the user notifying the automatic change

**Reply by BrunoBernardino** [reply]
Hi @Terkyz the first request isn't doable as it would require Padloc to interact with the item's website, in order to be able to change the password.

As for the email notification, I suppose we can consider it, but right now I think it would be mostly annoying, as people can simply go to the security report and fix the issues they have right now (specifically just for expired items, for example, if they have many other reused passwords and so on).

**Reply by MaKleSoft** [reply]
> As for the email notification, I suppose we can consider it

I'm afraid that's not possible either, since we don't have access to the expiry information on the server side since it's end-to-end encrypted along with everything else in the vault.

**Reply by BrunoBernardino** [reply]
My thinking was that we could initiate the request from the client, but you're correct that it could be sent "too late", if the user hadn't opened the app on the day of expiration. Yep, no go.

**Reply by Terkyz** [reply]
Is it possible to have built-in compatibility with temporary mail services like SimpleLogin, AnonAddy and Firefox Relay? It could be interesting to facilitate the administration of accounts created with temporary emails without depending many times on the pages of these services (which generally, without offending, are bad), and I also consider that if this were possible, there would be a button of "temporary mail" to generate an account using the temporary addresses already created with these services

**Reply by BrunoBernardino** [reply]
That's been requested and is being tracked at https://github.com/padloc/padloc/issues/596
